misc/python/materialize/data_ingest/transaction_def.py

```diff
@@ -43,3 +46,21 @@ def generate(self, fields: List[Field]) -> Iterator[Transaction]:
                     full_rowlist = []
         if full_rowlist:
             yield Transaction(full_rowlist)
+
+
+class RestartMz(TransactionDef):
+    composition: Composition
+    probability: float
+
+    def __init__(self, composition: Composition, probability: float):
+        self.composition = composition
+        self.probability = probability
+
+    def generate(self, fields: List[Field]) -> Iterator[Optional[Transaction]]:
+        if random.random() < self.probability:
+            self.composition.kill("materialized")
+            time.sleep(1)
+            self.composition.up("materialized")
+            # Wait until back up to prevent future queries from failing
```

Comment:
Have you seen the "future queries failing" situation? How does it manifest and why the `SELECT 1` would not fail in a similar fashion?

Reply:
c.up returns only after the healthcheck for Materialize has indicated that the service is operational. If there are any SQL errors after c.up, that would be a bug in the product and/or healthcheck.